### PR TITLE
gitkraken-on-premise-serverless 10.1.1 (new cask)

### PR DESCRIPTION
**Pull request by Moulick**
After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online gitkraken-serverless` is error-free.
- [x] `brew style --fix gitkraken-serverless` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new gitkraken-serverless` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask gitkraken-serverless` worked successfully.
- [x] `brew uninstall --cask gitkraken-serverless` worked successfully.

### CODE REVIEW

**Comment by Moulick** [comment]
Erm, why does SHA check fail if it passes on local? Locally I ran `HOMEBREW_NO_INSTALL_FROM_API=1 brew audit --new --strict --online gitkraken-serverless` and it gave no errors for the SHA for Intel build.

**Comment by Moulick** [comment]
It looks like locally brew audit only check the SHA for your particular arch. For _checksums_ I suppose it can download and compute that at-least?

**Comment by bevanjkay** [comment]
You would need to add `--arch=intel` to `brew audit` for that error to show.

File: Casks/g/gitkraken-serverless.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+cask "gitkraken-serverless" do
```
Comment (bevanjkay) [inline Casks/g/gitkraken-serverless.rb:1]: My only question is whether `gitkraken-serverless` is most suitable against alternatives `gitkraken-on-premise` or `gitkraken-on-premise-serverless`?
I'm not familiar with the product or the differences.

Comment (Moulick) [reply Casks/g/gitkraken-serverless.rb:1]: `gitkraken-on-premise-serverless` would probably be the correct name but everyone simply calls it GitKraken serverless. I'll rename to your suggestion

File: Casks/g/gitkraken-serverless.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  desc "Git client focusing on productivity"
+  homepage "https://www.gitkraken.com/git-client/on-premise"
+
+  livecheck do
+    url "https://www.gitkraken.com/download-on-premise-serverless"
+    regex(/Latest\srelease:\s(\d+(?:\.\d+)+)/i)
+  end
+
+  auto_updates true
+  conflicts_with cask: "gitkraken"
```
[the remark this answers is not in the document]
Comment (Moulick) [reply Casks/g/gitkraken-serverless.rb]: Never mind, it has to be a separate PR

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**